File: chazm-model/src/main/java/runtimemodels/chazm/model/entity/PmfEntity.java
package runtimemodels.chazm.model.entity;

import com.google.inject.assistedinject.Assisted;
import runtimemodels.chazm.api.entity.Pmf;
import runtimemodels.chazm.api.id.UniqueId;

import javax.inject.Inject;
import javax.validation.constraints.NotNull;

class PmfEntity extends AbstractEntity<Pmf> implements Pmf {

    @Inject
    PmfEntity(@NotNull @Assisted final UniqueId<Pmf> id) {
        super(id);
    }

    @Override
    public boolean equals(final Object object) {
        if (object instanceof Pmf) {
            final Pmf pmf = (Pmf) object;
            return super.equals(pmf);
        }
        return false;
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    public String toString() {
        return super.toString();
    }

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/entity/PmfEntity.kt
package runtimemodels.chazm.model.entity

import com.google.inject.assistedinject.Assisted
import runtimemodels.chazm.api.entity.Pmf
import runtimemodels.chazm.api.id.PmfId
import javax.inject.Inject

internal data class PmfEntity @Inject constructor(
    @Assisted override val id: PmfId
) : Pmf
